app/delivery/send_to_providers.py

```diff
@@ -104,13 +105,20 @@ def send_sms_to_provider(notification):
 
         else:
             try:
+                if current_app.config["FF_TEMPLATE_CATEGORY"]:
+                    template_category_id = template_dict.get("template_category_id")
+                    template_category = dao_get_template_category_by_id(template_category_id) if template_category_id else None
```

Comment:
Do you actually need to do this lookup?  Would just checking `template_category_id` for `null` be the same check without the db call?

Reply:
We need the category to get the sending vehicle from, but I've refactored and simplified this conditional:
```
template_category_id = template_dict.get("template_category_id")
if current_app.config["FF_TEMPLATE_CATEGORY"] and template_category_id is not None:
    sending_vehicle = dao_get_template_category_by_id(template_category_id).sms_sending_vehicle
else:
    sending_vehicle = None
```